dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/config/ListenableStateRouter.java

```diff
@@ -112,18 +122,45 @@ public BitList<Invoker<T>> doRoute(
         if (needToPrintMessage) {
             resultMessage = new StringBuilder();
         }
-        for (AbstractStateRouter<T> router : conditionRouters) {
-            invokers = router.route(invokers, url, invocation, needToPrintMessage, nodeHolder);
-            if (needToPrintMessage) {
-                resultMessage.append(messageHolder.get());
+
+        BitList<Invoker<T>> routeResult = invokers;
+        if (routerRule instanceof MultiDestConditionRouterRule
+                || routerRule.getVersion() != null && routerRule.getVersion().startsWith(RULE_VERSION_V31)) {
+            boolean trafficDisable = false;
+            for (MultiDestConditionRouter<T> multiDestConditionRouter : multiDestConditionRouters) {
+                routeResult = multiDestConditionRouter.route(invokers, url, invocation, needToPrintMessage, nodeHolder);
```

Comment:
Should clone `invokers` here first

Reply:
I don't know if cloning is necessary here; the v3.0 logic just passes the filtering on invokers itself.

Reply:
In the route method, the value in invokers is not modified directly, and there are three cases:
1.Return invokers directly
2. return BItlist.empty
3. The invokers will be cloned and returned

so I don't think it is necessary to clone first here, just keep the same as v3.0